tests/test_nodes_utils.py

```diff
@@ -40,3 +48,71 @@ def test_consume_common_prefix(left, right, expected):
     expected_b = (expected[0], expected[2], expected[1])
     assert actual_a == expected
     assert actual_b == expected_b
+
+
+@pytest.mark.parametrize(
+    'node',
+    (
+        (b''),
+        (None),
+        (b"\x00\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x01\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b"\x01\x02\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p\xc5\xd2F\x01\x86\xf7#<\x92~}\xb2\xdc\xc7\x03\xc0\xe5\x00\xb6S\xca\x82';{\xfa\xd8\x04]\x85\xa4p"),  # noqa: E501
+        (b'\x02'),
+    ),
+)
+def test_binary_trie_node_parsing_with_invalid_input(node):
+    try:
+        parse_node(node)
+    except InvalidNode:
```

Comment:
This test would currently pass if any of the test cases above did not raise `InvalidNode`

The way to test this type of thing using pytest is using `pytest.raises`

```python
with pytest.raises(InvalidNode):
    parse_node(node)
```

Tests written this way will fail if the provided exception is not raised.